File: web-node/scripts/build-logo-svg.mjs
import { mkdirSync, readFileSync, writeFileSync } from "fs";
import { dirname, join } from "path";
import { fileURLToPath } from "url";

const root = join(dirname(fileURLToPath(import.meta.url)), "..");
const pngPath = join(root, "logo_converta.png");
const data = readFileSync(pngPath);
const width = data.readUInt32BE(16);
const height = data.readUInt32BE(20);
const b64 = data.toString("base64");

const svg =
  `<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" ` +
  `width="${width}" height="${height}" viewBox="0 0 ${width} ${height}" ` +
  `role="img" aria-label="Converta Ads by Retail Media">` +
  `<title>Converta Ads by Retail Media</title>` +
  `<image width="${width}" height="${height}" preserveAspectRatio="xMidYMid meet" ` +
  `xlink:href="data:image/png;base64,${b64}"/></svg>`;

const targets = [
  join(root, "logo_converta.svg"),
  join(root, "web-js", "assets", "logo_converta.svg"),
  join(root, "web-node", "public", "assets", "logo_converta.svg"),
  join(root, "web-php", "public", "assets", "logo_converta.svg"),
];

for (const file of targets) {
  mkdirSync(dirname(file), { recursive: true });
  writeFileSync(file, svg, "utf8");
  console.log("OK", file, `${width}x${height}`);
}

File: build_logo_svg.py
"""Gera logo_converta.svg a partir do PNG (aparência idêntica via embed)."""
from __future__ import annotations

import base64
import struct
from pathlib import Path

ROOT = Path(__file__).resolve().parent
PNG = ROOT / "logo_converta.png"

TARGETS = [
    ROOT / "logo_converta.svg",
    ROOT / "web-js" / "assets" / "logo_converta.svg",
    ROOT / "web-node" / "public" / "assets" / "logo_converta.svg",
    ROOT / "web-php" / "public" / "assets" / "logo_converta.svg",
]


def main() -> None:
    data = PNG.read_bytes()
    width, height = struct.unpack(">II", data[16:24])
    b64 = base64.b64encode(data).decode("ascii")
    svg = (
        f'<svg xmlns="http://www.w3.org/2000/svg" '
        f'xmlns:xlink="http://www.w3.org/1999/xlink" '
        f'width="{width}" height="{height}" viewBox="0 0 {width} {height}" '
        f'role="img" aria-label="Converta Ads by Retail Media">\n'
        f'  <title>Converta Ads by Retail Media</title>\n'
        f'  <image width="{width}" height="{height}" preserveAspectRatio="xMidYMid meet" '
        f'xlink:href="data:image/png;base64,{b64}"/>\n'
        f"</svg>\n"
    )
    for path in TARGETS:
        path.parent.mkdir(parents=True, exist_ok=True)
        path.write_text(svg, encoding="utf-8")
        print(f"OK {path} ({width}x{height})")


if __name__ == "__main__":
    main()
